transports/janus_rabbitmq.c

```diff
@@ -333,12 +323,22 @@ int janus_rabbitmq_init(janus_transport_callbacks *callback, const char *config_
 			/* Something went wrong... */
 			JANUS_LOG(LOG_FATAL, "Got error %d (%s) trying to launch the RabbitMQ outgoing thread...\n", error->code, error->message ? error->message : "??");
 			g_free(rmq_client);
+			janus_config_destroy(config);
 			return -1;
 		}
 		janus_mutex_init(&rmq_client->mutex);
 		/* Done */
 		JANUS_LOG(LOG_INFO, "Setup of RabbitMQ integration completed\n");
 	}
+	g_free(rmqhost);
+	if(to_janus)
+		g_free((char *)to_janus);
+	if(from_janus)
+		g_free((char *)from_janus);
+	if(to_janus_admin)
+		g_free((char *)to_janus_admin);
+	if(from_janus_admin)
+		g_free((char *)from_janus_admin);
```

Comment:
Hi guys, could someone tell me the gain (CPU...) we expect by replacing
sleeps with conditions (wait/signal) ?
Just curious :)

On Thu, Dec 3, 2015 at 8:25 PM, Pierce Lopez <notifications@github.com>
wrote:

> In transports/janus_rabbitmq.c
> <https://github.com/meetecho/janus-gateway/pull/384#discussion_r46597956>:
>
> >  			return -1;
> >  		}
> >  		janus_mutex_init(&rmq_client->mutex);
> >  		/* Done */
> >  		JANUS_LOG(LOG_INFO, "Setup of RabbitMQ integration completed\n");
> >  	}
> > +	g_free(rmqhost);
> > +	if(to_janus)
> > +		g_free((char *)to_janus);
> > +	if(from_janus)
> > +		g_free((char *)from_janus);
> > +	if(to_janus_admin)
> > +		g_free((char *)to_janus_admin);
> > +	if(from_janus_admin)
> > +		g_free((char *)from_janus_admin);
>
> sounds good to me
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/meetecho/janus-gateway/pull/384/files#r46597956>.
>